### DISCUSSION TITLE
Blur Red into Neighboring Color

### DISCUSSION THREAD

**Discussion by trythisone35** [ImageMagick/ImageMagick#6632] [Help]
### ImageMagick version

7.1.1-15-Q16-HDRI-x64

### Operating system, version and so on

Windows 10

### Description

I have a series of images that have a series of callouts and annotations in red.  I'd like to have the red blend into the surrounding colors.  I originally tried transparent but couldn't get that to work.  I was using the following:

magick bear.jpg -transparent red filtered.jpg

I also realized that transparent would be a problem with red artwork which is when I started to wonder if the red could blur and dissolve into neighboring colors.  
![bear](https://github.com/ImageMagick/ImageMagick/assets/79946966/9f726590-8d75-48cd-9f0d-17a0b649f4f0)
 

One other note is that I might be able to dictate the precise red going forward if that makes it easier.

Thanks in advance.

**Comment by fmw42** [comment]
Add -fuzz XX% before -transparent.  Adjust the XX% as desired or needed

**Reply by trythisone35** [reply]
Am I missing something in the command?  

magick bear.jpg -fuzz 10% -transparent red output.jpg

I'm unable to get any transparency at all.

**Comment by snibgo** [comment]
>  I'd like to have the red blend into the surrounding colors. 

What does that mean? Perhaps you want to change the colour of the "red" pixels so they are more similar to the backgound (which is 80% gray). Or the opposite, do you want to change adjacent gray pixels to be somewhat red? How does transparency help? What is the "red" colour? Is the fish red? Is the bear red?

**Reply by trythisone35** [reply]
Sorry for the confusion.

Let me phrase the need a different way.  In the bear image you will see red text, red lines, and some red arrows.   I want those to no longer be visible.  

When invoking the following command:   magick bear.jpg -transparent red output.png 

It had no impact on the resulting image.  The red still persisted in the image.

**Comment by fmw42** [comment]
First, JPG does not support transparency.  So you should not use output format of JPG when doing -transparent.

Second, your variation in red is too great (due to antialiased text) such that if you put a large enough -fuzz value , it will also change the orange of the bear.
